micronaut-maven-jib-integration/src/main/java/io/micronaut/maven/jib/JibMicronautExtension.java

```diff
@@ -121,8 +121,10 @@ public static List<String> buildProjectFnEntrypoint() {
 
     public static String determineProjectFnVersion(String javaVersion) {
         int majorVersion = Integer.parseInt(javaVersion.split("\\.")[0]);
-        if (majorVersion >= 17) {
-            return "jre17-latest";
+        if (majorVersion <= 21 && majorVersion > 17) {
+            return "21-jre";
+        } else if (majorVersion <= 17) {
```

Comment:
Earlier JDK versions will never get to this point

```suggestion
        } else if (majorVersion == 17) {
```